src/components/CollectionImage/CollectionImage.tsx

```diff
@@ -1,47 +1,76 @@
 import * as React from 'react'
 import { Loader } from 'decentraland-ui'
 import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import CollectionProvider from 'components/CollectionProvider'
 import { Item } from 'modules/item/types'
 import ItemImage from 'components/ItemImage'
 import { Props } from './CollectionImage.types'
 import './CollectionImage.css'
 
+const MAX_IMAGES_TO_SHOW = 4
+
 export default class CollectionImage extends React.PureComponent<Props> {
   renderItemRow(items: Item[]) {
     return items.map((item, index) => <ItemImage key={index} item={item} />)
   }
 
-  render() {
-    let { items, isLoading } = this.props
-
+  renderItemRows(items: Item[]) {
     const firstItemRow = items.slice(0, 2)
-    const secondItemRow = items.slice(2, 4)
+    const secondItemRow = items.slice(2, MAX_IMAGES_TO_SHOW)
     const itemRowStyle = { height: secondItemRow.length ? '50%' : '100%' }
-
     return (
-      <div className="CollectionImage is-image">
-        {isLoading ? (
-          <div className='item-row loading'>
+      <>
+        {firstItemRow.length > 0 ? (
+          <div className="item-row" style={itemRowStyle}>
+            {this.renderItemRow(firstItemRow)}
+          </div>
+        ) : null}
+        {secondItemRow.length > 0 ? (
+          <div className="item-row" style={itemRowStyle}>
+            {this.renderItemRow(secondItemRow)}
+          </div>
+        ) : null}
+      </>
+    )
+  }
+
+  render() {
+    const { items, itemCount, collectionId, isLoading } = this.props
+
+    if (isLoading || itemCount === null) {
```

Comment:
Item count is typed as undefined or number, it should never take null as its value.